permission/core/permissions.go

```diff
@@ -13,10 +13,39 @@ import (
 	"github.com/ethereum/go-ethereum/params"
 )
 
-// check if a given node is permissioned to connect to the change
+type FileBasedPermissioning struct {
+	PermissionFile  string
+	DissallowedFile string
+}
+
+var defaultFileBasedPermissioning = FileBasedPermissioning{
+	PermissionFile:  params.PERMISSIONED_CONFIG,
+	DissallowedFile: params.BLACKLIST_CONFIG,
+}
+
+func NewFileBasedPermissoningWithPrefix(prefix string) FileBasedPermissioning {
+	return FileBasedPermissioning{
+		PermissionFile:  prefix + "-" + params.PERMISSIONED_CONFIG,
+		DissallowedFile: prefix + "-" + params.BLACKLIST_CONFIG,
+	}
+}
+
 func IsNodePermissioned(nodename string, currentNode string, datadir string, direction string) bool {
+	return defaultFileBasedPermissioning.IsNodePermissioned(nodename, currentNode, datadir, direction)
+}
+
+func isNodeBlackListed(nodeName, dataDir string) bool {
```

Comment:
we made PR for inclusive terms, could it be Disallowed?

Reply:
fixed